### PR TITLE
[wasm] Simd refactoring

### PR DESCRIPTION
**Pull request by radekdoulik**
Rename few opcodes to contain `_SIMD_`.

Use more SimdIntrinsic fields to simplify the code.

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -4711,8 +4711,8 @@ static SimdIntrinsic packedsimd_methods [] = {
 	{SN_Bitmask, OP_WASM_SIMD_BITMASK},
 	{SN_CompareEqual},
 	{SN_CompareNotEqual},
-	{SN_ConvertNarrowingSignedSaturate, OP_XOP_X_X_X},
-	{SN_ConvertNarrowingUnsignedSaturate, OP_XOP_X_X_X},
+	{SN_ConvertNarrowingSignedSaturate, OP_XOP_X_X_X, 0},
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:4714]: This is probably not needed, the unset fields are 0.

File: src/mono/mono/mini/mini-ops.h [hunk 2]
```diff
@@ -1757,17 +1757,17 @@ MINI_OP3(OP_ARM64_SQRDMLSH_SCALAR, "arm64_sqrdmlsh_scalar", XREG, XREG, XREG, XR
...
+MINI_OP(OP_SIMD_FCVTL2, "convert_to_higher_precision_2", XREG, XREG, NONE)
+MINI_OP(OP_SIMD_USHLL, "unsigned_shift_left_long", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_USHLL2, "unsigned_shift_left_long_2", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_SSHLL, "signed_shift_left_long", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_SSHLL2, "signed_shift_left_long_2", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_SHL, "shl", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_SSHR, "sshr", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_USHR, "ushr", XREG, XREG, IREG)
+MINI_OP3(OP_SIMD_USRA, "usra", XREG, XREG, XREG, IREG)
+MINI_OP3(OP_SIMD_SSRA, "ssra", XREG, XREG, XREG, IREG)
```
Comment (kg) [inline src/mono/mono/mini/mini-ops.h:1770]: shouldn't the string versions also start with simd_? like for the wasm ones below

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -4852,25 +4849,16 @@ emit_wasm_supported_intrinsics (
...
 				return emit_simd_ins (cfg, klass, type_to_expand_op (etype->type), args [0]->dreg, -1);
 			}
-			case SN_Dot:
-				return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_X, INTRINS_WASM_DOT, -1, fsig, args);
-			case SN_Shuffle:
-				return emit_simd_ins_for_sig (cfg, klass, OP_WASM_SIMD_SHUFFLE, -1, -1, fsig, args);
-			case SN_Swizzle:
-				return emit_simd_ins_for_sig (cfg, klass, OP_WASM_SIMD_SWIZZLE, -1, -1, fsig, args);
 		}
+		if (op != 0)
```
Comment (kg) [inline src/mono/mono/mini/simd-intrinsics.c:4858]: Maybe add a comment specifying that this is a default fallback

File: src/mono/mono/mini/simd-intrinsics.c [hunk 4]
```diff
@@ -4796,54 +4796,51 @@ emit_wasm_supported_intrinsics (
...
 						break;
 				case MONO_TYPE_I4:
-						intrins = INTRINS_WASM_NARROW_SIGNED_V8;
+						c0 = INTRINS_WASM_NARROW_SIGNED_V8;
 						break;
 				}
-				if (intrins != -1)
-						return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_X, intrins, arg0_type, fsig, args);
+				if (c0 != 0)
+						break;
```
Comment (kg) [inline src/mono/mono/mini/simd-intrinsics.c:4823]: Might also be good to have comments indicating that these breaks are explicitly intended to use the fallback, like a fall-through

**Review by kg — APPROVED** [review]

**Comment by radekdoulik** [comment]
The failing builds are unrelated

**Pull request merged**